pkg/utils/retry.go

```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"time"
+)
+
+// Retry will make a given number of attempts to run the provided function
+// with exponential backoff
+func Retry(attempts int, sleep time.Duration, fn func() error) error {
+	if err := fn(); err != nil {
+		if s, ok := err.(stop); ok {
+			// Return the original error for later checking
+			return s.error
+		}
+
+		if attempts--; attempts > 0 {
+			time.Sleep(sleep)
```

Comment:
I would also agrue that there should be no need for a sleep, this should be done with a channel